### PR TITLE
component/legislation-timeline

### LINKED ISSUE #175: [f_legislationtracking] Legislation history component
Implement the Legislation history component of the legislation details page (`/matters/:id`).

See the figma lofi mocks ups https://www.figma.com/file/VPeHF2SVDn6FneLLl9bihB/Legislation-Tracking-(CDP)?node-id=137%3A350
image

- The timeline should be in reverse order so that the latest history item is first
- The component is an accordion -- allow users to toggle the visibility of the timeline

### PR DESCRIPTION
**Pull request by BrianL3**
### Link to Relevant Issue

This pull request resolves #175 

### Description of Changes

This completes one of the legislation tracking feature subcomponents: the legislation's history. 

This ended up being slightly different from the [proposed figma](https://www.figma.com/file/VPeHF2SVDn6FneLLl9bihB/Legislation-Tracking-(CDP)?node-id=137%3A350).  Instead of a plain unordered html list, I used colored Dots to indicate the decision reached (if any) at any particular EventMinutesItem in the timeline. This increased readability at a glance.

In addition, the vote details available in the expanded view leverage the already-created component VoteGraphicDistribution, meaning users can click into the votes for specific names of councilors. 

Legislative action is not currently available, which is unfortunate because that would have provided a neat title for each node on the timeline. The date of the associated Event is used instead, which at least provides some context. 

### Link to Forked Storybook Site

[storybook](https://brianl3.github.io/cdp-frontend/?path=/story/library-details-legislation-history--history)

### CODE REVIEW

**Review by tohuynh — CHANGES_REQUESTED** [review]
Overall looks good.

I have some quick fixes in the comments.

---
- I think we should link the event in each of the timeline items that occur during an event. (Opton: Use the date as the `a` element)
- I think we should have a show/hide votes in a `details`. Basically have a `details` below the date that shows or hides the votes. Generally speaking, element that can take user interaction should be tabable by keyboard. Clicking on the dot hide/show the votes, but the dot itself can't be tabbed to. With `details` at least the user can tab to `hide/show votes`.
- I'm OK with leaving out the list of attachments for each timeline item (if it has them).
- Can we have the date and votes component line up vertically on the left? Current the votes component looks a little more left than the date .

File: src/components/Details/Legislation/VoteDistributionGraphic.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "{number}",
+    `${against.length}`
+  );
+  const votesAbstainedLabel: string = strings.number_non_voting.replace(
+    "{number}",
+    `${abstained.length}`
+  );
+
+  const voteBars: ReactNode[] = [
+    <VoteBar
```
[the remark this answers is not in the document]
Comment (tohuynh) [reply src/components/Details/Legislation/VoteDistributionGraphic.tsx:29]: image

**Comment by BrianL3** [comment]
> I think we should have a show/hide votes in a details. Basically have a details below the date that shows or hides the votes. Generally speaking, element that can take user interaction should be tabable by keyboard. Clicking on the dot hide/show the votes, but the dot itself can't be tabbed to. With details at least the user can tab to hide/show votes.


I am having some difficulty using a `details` here, storybook is telling me: Ensure controls are not nested as they are not announced by screen readers.

**Comment by tohuynh** [comment]
> I am having some difficulty using details here, storybook is telling me: Ensure controls are not nested as they are not announced by screen readers.

It shouldn't be a problem to use have details inside another details? I use it on the person page, in `PersonRoles`.

File: src/components/Details/Legislation/VoteBar.tsx [hunk 2]
```diff
@@ -37,22 +39,12 @@ const VoteBar: FC<VoteBarProps> = ({
...
-              .toLowerCase()
-              .replace(" ", "_")
-              .replace("-", "_")
-              .replace("(", "")
-              .replace(")", "")
-          ]
-        }
-      </b>
+      <DecisionResult result={vote.decision} />
+      <Spacer style={{ flex: 4 }} />
```
Comment (tohuynh) [inline src/components/Details/Legislation/VoteBar.tsx:47]: Let's remove the Spacer, on mobile it looks congested

try

```
const Votes = styled.div({
  marginTop: 4,
  display: "grid",
  gridTemplateColumns: "auto auto",
  justifyContent: "space-between",
  [`@media (min-width:${screenWidths.tablet})`]: {
    justifyContent: "start",
    gridTemplateColumns: "0.3fr 1fr"
  },
})
```

```
<Votes key={`${i}_${vote.id}_vote`}>
      <Link to={`/people/${vote.id}`} style={{ marginLeft: 12 }}>
        {vote.person?.name}
      </Link>
      <DecisionResult result={vote.decision} />
    </Votes>
```

File: src/components/Details/Legislation/LegislativeHistoryNode.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+const ReferenceFrame = styled.div({ position: "relative" });
+const ExpandingContainer = styled.div({
+  margin: MARGIN,
+  display: "flex",
+  flexDirection: "row",
+  gap: 32,
+});
+const TextContainer = styled.div({ flex: 1 });
+const VotingGraphicContainer = styled.div({ flex: 1, marginTop: MARGIN, marginLeft: MARGIN * 2 });
```
[the remark this answers is not in the document]
Comment (BrianL3) [reply src/components/Details/Legislation/LegislativeHistoryNode.tsx:42]: I made some changes for mobile.

**Review by JacksonMaxfield — COMMENTED** [review]
We close!!! Nice work.

I like the idea of showing the overall matter decision color somewhere but _just_ the color isn't enough for me. Can we make those "progress dots" all grey / black like normal but include a "Passed" with the checkmark icon or rejected with the X icon or etc. Below the meeting date / link?

In addition to votes should be the linked event minutes items. The point of those is to say "an amendment was introduced in this meeting" or "a report was given in this earlier meeting" -- like the provenance of the matter basically.

**Comment by BrianL3** [comment]
> In addition to votes should be the linked event minutes items. The point of those is to say "an amendment was introduced in this meeting" or "a report was given in this earlier meeting" -- like the provenance of the matter basically.

Could you clarify what you mean by this? The entire Node, all the info from decision to date, is the EventMinutesItem.

**Review by JacksonMaxfield — APPROVED** [review]
Looks good to me! Will wait for @tohuynh to approve but seems like my task for spring break is to fix the index pipelines + make the matter index pipeline and then this whole feature is ready to go :tada:

File: src/components/Details/Legislation/LegislativeHistoryNode.tsx [hunk 4]
```diff
@@ -13,54 +14,57 @@ import { Dot, DOT_SIZE } from "../../Shared/Dot";
...
+    margin: MARGIN,
+    display: "flex",
+    flexDirection: "row",
+    gap: MARGIN * 3,
+  });
+  const TextContainer = styled.div({ flex: 1 });
+  const VotingGraphicContainer = styled.div<{ marginLeft: number }>((props) => ({
+    flex: 1,
+    marginTop: MARGIN,
+    marginLeft: props.marginLeft,
```
Comment (tohuynh) [inline src/components/Details/Legislation/LegislativeHistoryNode.tsx:57]: Just a suggestion: you could add the media query here
```
const VotingGraphicContainer = styled.div({
    flex: 1,
    marginTop: MARGIN,
   marginLeft: 0, //on mobile
   [`@media (min-width:${screenWidths.tablet})`]: {
         marginLeft: MARGIN * 2 //on tablet or above
  },

```
I think this is more succient to keep the styling logic in one place.

File: src/components/Details/Legislation/LegislativeHistoryNode.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                  marginLeft: MARGIN,
+                  display: "flex",
+                  flex: 1,
+                  alignItems: "center",
+                }}
+              />
+            )}
+          </SummaryInfo>
+          <LazyFetchDataContainer
+            data="legislativeHistoryNodeDataState"
```
Comment (tohuynh) [inline src/components/Details/Legislation/LegislativeHistoryNode.tsx:127]: the `data` prop is used to display message like
- `Fetching <data>...`
- `No <data> found.`

Could you change the text to something more appropriate?

**Review by tohuynh — APPROVED** [review]
Thanks for making the changes.

I have just one more minor text request.

File: src/components/Details/Legislation/LegislativeHistoryNode.tsx (hunk 5 above)
Comment (BrianL3) [reply src/components/Details/Legislation/LegislativeHistoryNode.tsx:127]: Yeah. I added a comment to the data prop, for clarity.

**Pull request merged**